dynamodb/dynamodb_test.go

```diff
@@ -0,0 +1,177 @@
+package dynamodb
+
+import (
+	"testing"
+
+	bindings "github.com/andream16/aws-sdk-go-bindings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/pkg/errors"
+)
+
+type mockDynamoDBClient struct {
+	dynamodbiface.DynamoDBAPI
+}
+
+func (mockDynamoDBClient) PutItem(in *dynamodb.PutItemInput) (*dynamodb.PutItemOutput, error) {
+	return nil, nil
+}
+
+func (mockDynamoDBClient) GetItem(in *dynamodb.GetItemInput) (*dynamodb.GetItemOutput, error) {
+	return &dynamodb.GetItemOutput{
+		Item: map[string]*dynamodb.AttributeValue{
+			"keyName": &dynamodb.AttributeValue{
+				S: aws.String("someValue"),
+			},
+		},
+	}, nil
+}
+
+type mockFailingDynamoDBClient struct {
+	dynamodbiface.DynamoDBAPI
+}
+
+func (mockFailingDynamoDBClient) PutItem(*dynamodb.PutItemInput) (*dynamodb.PutItemOutput, error) {
+	return nil, errors.New("some error")
+}
+
+func (mockFailingDynamoDBClient) GetItem(*dynamodb.GetItemInput) (*dynamodb.GetItemOutput, error) {
+	return nil, errors.New("some error")
+}
+
+type item struct {
+	KeyName string
+}
+
+func TestDynamoDB_PutItem(t *testing.T) {
+
+	t.Run("should return an error because table name is empty", func(t *testing.T) {
+
+		dynamoDB := &DynamoDB{}
+
+		err := dynamoDB.PutItem("", nil)
+		if bindings.ErrInvalidParameter != errors.Cause(err) {
+			t.Fatalf("expected error %s, got %s", bindings.ErrInvalidParameter, err)
+		}
+
+	})
+
+	// This case is left untested since dynamodbattribute.MarshalMap does not return any error in aws-sdk-go v1.19.14
+	t.Run("should return an error because there was an error marshaling item into map", func(t *testing.T) {})
```

Comment:
It's fine removing it. Remember that we are testing in isolation & testing our stuff. If things like that are difficult to test, just don't do it. We are assuming that mr Bezos tested it well since it's an exported method.